### ISSUE TITLE
When trying to use the blueprint I receive Error: list index out of range

### ISSUE THREAD

**Issue by vcdx71** [music-assistant/voice-support#1]
I'm trying to use this blueprint. When I ask assist to play an artist or a song I get a response back of "Done". Checking the traces for the automation shows Error: list index out of range.

full trace is attached.

HA Version: 2024.12.5
OS: 14.1
MA: 2.3.3 (running in a docker container on Unraid, same L2 network)

[trace automation.music_assistant_voice_official 2024-12-23T04_49_35.090136+00_00.json](https://github.com/user-attachments/files/18225235/trace.automation.music_assistant_voice_official.2024-12-23T04_49_35.090136%2B00_00.json)

**Comment by marcelveldt** [comment]
I've updated the blueprint just now - can you try it again ?

**Comment by vcdx71** [comment]
Thanks, this is closer. It now plays but it's always playing on the default player, it's not going to the area I tell it to.

![trace.zip](https://github.com/user-attachments/files/18296823/trace.zip)

**Comment by tesnos6921** [comment]
I found that assigning the devices created by the Music Assistant integration to their proper areas + removing line 110 to not check the friendly_name was enough to make it work for me

**Comment by vcdx71** [comment]
> I found that assigning the devices created by the Music Assistant integration to their proper areas + removing line 110 to not check the friendly_name was enough to make it work for me

Thanks, I already had the areas setup but removing line 110 fixed it for me as well.

**Comment by eric10k93** [comment]
Which specific blueprint was this? Seems like they've all been changed quite a bit since this initial problem... I was actually getting this exact error message, but after a lot of digging around, I found that my problem related to the media provider I was using, Plex.  I haven't had a lot of luck getting either of the automation blueprints to work, but WAS able to get the script to work, which I think is my preferred method, since I like using an LLM with local controls.

So my issue causing this error message was related to the fact that I was trying to use 'radio mode' with the Plex provider... It's apparently not supported... By simply changing 'radio mode' from 'Always' to 'Never' it stopped generating the 'index out of range' error and now the script works for me.

**Comment by TheFes** [comment]
> EDIT: To be clear, I was able to get the 'Local Only' blueprint to work ok, but it always plays on the default provider... It never plays on any other device or area requested.

The local automation only works with the actual and exact name of the device or area, not with aliases.


> As for the 'LLM with Local' blueprint, I don't think it plays nice with Google Gemini, as it just won't work no matter what I try... I think it has something to do with the conversation agent. The blueprint says it needs an LLM with local controls disabled, but Google Gemini doesn't even have that as an option at the 'conversation agent' level... That's defined when setting it up as a 'Voice Assistant', but the blueprint doesn't point to a specific Voice Assistant provider... It's requesting a conversation agent.

This is not a setting in the voice pipeline, it's a setting on the conversation agent itself

![Image](https://github.com/user-attachments/assets/8169d784-6b1d-4415-bd76-b5ceb5972711)

With the setting marked by the arrow you allow the LLM to control your house, the 2nd option is created specifically for when that setting is set not `No control`

**Comment by eric10k93** [comment]
> The local automation only works with the actual and exact name of the device or area, not with aliases.
Yeah, I was using exact names, not aliases.  It still kept falling back to the default device.  I couldn't really tell why, when stepping through the Traces seemed to indicate it was correctly finding the device I was requesting.

> This is not a setting in the voice pipeline, it's a setting on the conversation agent itself

> With the setting marked by the arrow you allow the LLM to control your house, the 2nd option is created specifically for when that setting is set not `No control`

I noticed this dialog is also available from the integration configuration menu, as you pointed out.  I initially only found this when setting up a new voice pipeline and selecting Google AI as the conversation agent.

**Comment by TheFes** [comment]
@eric10k93 If you have a trace of an occurrence where you provided a target in the voice command, but it played on the default player I can investigate a bit further. But if you are fine with using option 3 and it all works now, that's fine for me as well :)

**Comment by nikito** [comment]
Closing as no activity in over a month. Please reopen if issue is still occurring and further action is needed! :)
